test/unit/processing/settlement/schema.test.js

```diff
@@ -0,0 +1,116 @@
+const schema = require('../../../../app/processing/settlement/schema')
+let settlement
+
+describe('settlement schema', () => {
+  beforeEach(() => {
+    settlement = {
+      paymentRequestId: 1,
+      invoiceNumber: 'S12345671234567V001',
+      reference: 'PY1234567',
+      settled: true,
+      settlementDate: new Date(2022, 9, 12),
+      value: 50000
+    }
+  })
+
+  test('should not error if valid', () => {
+    const result = schema.validate(settlement)
+    expect(result.error).toBeUndefined()
+  })
+
+  test('should error if paymentRequestId null', () => {
+    settlement.paymentRequestId = null
+    const result = schema.validate(settlement)
+    expect(result.error).toBeDefined()
+  })
+
+  test('should error if paymentRequestId undefined', () => {
+    settlement.paymentRequestId = undefined
+    const result = schema.validate(settlement)
+    expect(result.error).toBeDefined()
+  })
+
+  test('should error if invoiceNumber null', () => {
+    settlement.invoiceNumber = null
+    const result = schema.validate(settlement)
+    expect(result.error).toBeDefined()
+  })
+
+  test('should error if invoiceNumber undefined', () => {
+    settlement.invoiceNumber = undefined
+    const result = schema.validate(settlement)
+    expect(result.error).toBeDefined()
+  })
+
+  test('should error if invoiceNumber not a string', () => {
+    settlement.invoiceNumber = 1
+    const result = schema.validate(settlement)
+    expect(result.error).toBeDefined()
+  })
+
+  test('should error if invoiceNumber not a first payment', () => {
+    settlement.invoiceNumber = 'S12345671234567V002'
```

Comment:
could create a new settlement object in the `mock-settlement` file which is not a 1st payment and use it here -- would require reference changes elsewhere in the tests to point to the 1st settlement